python-backend/app/api/permits/permit_amendment/models/permit_amendment.py

```diff
@@ -0,0 +1,100 @@
+from sqlalchemy.orm import validates
+from sqlalchemy.schema import FetchedValue
+from app.extensions import db
+
+from . import permit_amendment_status_code, permit_amendment_type_code
+from ....utils.models_mixins import AuditMixin, Base
+
+
+class PermitAmendment(AuditMixin, Base):
+    __tablename__ = 'permit_amendment'
+    permit_amendment_id = db.Column(db.Integer, primary_key=True, server_default=FetchedValue())
+    permit_amendment_guid = db.Column(UUID(as_uuid=True))
+    permit_id = db.Column(db.Integer, db.ForeignKey('permit.permit_id'), nullable=False)
+    received_date = db.Column(db.DateTime, nullable=False)
+    issue_date = db.Column(db.DateTime, nullable=False)
+    authorization_end_date = db.Column(db.DateTime, nullable=False)
+    permit_amendment_status_code = db.Column(
+        db.String(3), db.ForeignKey('permit_amendment_status_code.permit_amendment_status_code'))
+    permit_amendment_type_code = db.Column(
+        db.String(3), db.ForeignKey('permit_amendment_type_code.permit_amendment_type_code'))
+
+    def json(self):
+        return {
+            'permit_amendment_id':
+            self.permit_amendment_id,
+            'permit_amendment_guid':
+            str(self.permit_amendment_guid),
+            'permit_id':
+            self.permit_id,
+            'permit_amendment_status_code':
+            self.permit_amendment_status_code,
+            'permit_amendment_type_code':
+            self.permit_amendment_type_code,
+            'received_date':
+            self.received_date.isoformat() if self.received_date else None,
+            'issue_date':
+            self.issue_date.isoformat() if self.issue_date else None,
+            'authorization_end_date':
+            self.authorization_end_date.isoformat() if self.authorization_end_date else None
+        }
+
+    @classmethod
+    def find_by_permit_amendment_id(cls, _id):
+        return cls.query.filter_by(permit_amendment_id=_id).first()
+
+    @classmethod
+    def find_by_permit_id(cls, _id):
+        return cls.query.filter_by(permit_id=_id)
+
+    @validates('permit_amendment_status_code')
+    def validate_status_code(self, key, permit_amendment_status_code):
+        if not permit_amendment_status_code:
+            raise AssertionError('Permit amendment status code is not provided.')
+        return permit_amendment_status_code
+
+    @validates('permit_amendment_type_code')
+    def validate_type_code(self, key, permit_amendment_type_code):
+        if not permit_amendment_type_code:
+            raise AssertionError('Permit amendment type code is not provided.')
+        return permit_amendment_type_code
+
+    @validates('permit_id')
+    def validate_permit_id(self, key, permit_id):
+        if not permit_id:
+            raise AssertionError('Permit id is not provided.')
+        return permit_id
+
+    @validates('received_date')
+    def validate_received_date(self, key, received_date):
+        if not authorization_end_date:
```

Comment:
I believe this is a typo but I think this should be received_date.

Reply:
Thanks. Changed :)